onnxruntime/core/providers/cuda/tensor/resize_impl.cu

```diff
@@ -359,6 +444,88 @@ size_t CalcResizeBufferSize(const onnxruntime::UpsampleMode upsample_mode,
   return 0;
 }
 
+template <typename T>
+void ResizeNearestImpl(
+    const int rank,
+    CudaKernel::CudaAsyncBuffer<int64_t>& input_shape,
+    CudaKernel::CudaAsyncBuffer<int64_t>& output_shape,
+    CudaKernel::CudaAsyncBuffer<int64_t>& input_strides,
+    CudaKernel::CudaAsyncBuffer<fast_divmod>& output_div_pitches,
+    CudaKernel::CudaAsyncBuffer<float>& scales_vals,
+    CudaKernel::CudaAsyncBuffer<float>& roi_vals,
+    const T* input_data,
+    T* output_data,
+    const size_t N,
+    bool extrapolation_enabled,
+    const T extrapolation_value,
+    float cubic_coeff_a,
+    CudaFunctionOriginalCoordinate transform_coordinate,
+    CudaFunctionNearestPixel calc_nearest_pixel,
+    int64_t* prefix_dim_sum,
+    NearestMappingInfo* dims_mapping) {
+  int blocksPerGrid = (int)(ceil(static_cast<float>(N) / GridDim::maxThreadsPerBlock));
+
+  bool could2d = rank >= 2 &&
+                 transform_coordinate != GetDeviceOriginalCoordinateFunc(ResizeCoordinateTransformationMode::TF_CROP_AND_RESIZE) &&
+                 std::all_of(scales_vals.CpuPtr(), scales_vals.CpuPtr() + (rank - 2), [](float v) { return v == 1.0; });
+  if (could2d) {
+    int64_t output_height = output_shape.CpuPtr()[rank - 2];
+    int64_t output_width = output_shape.CpuPtr()[rank - 1];
+    fast_divmod div_output_image = (rank > 2) ? output_div_pitches.CpuPtr()[rank - 3] : fast_divmod(output_height * output_width);
+    int blocksPerDimsMappingGrid = (int)(ceil((output_height + output_width) / 32.0));
+
+    _ResizeNearestMappingKernel2D<T><<<blocksPerDimsMappingGrid, 32, 0>>>(
```

Comment:
why 32?

Reply:
As data could be small, do not want it in big block since where inner wraps may chained.